extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -403,10 +404,10 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 	}
 
 	public async getDatabaseAssessments(targetType: MigrationTargetType[]): Promise<ServerAssessment> {
-		const ownerUri = await azdata.connection.getUriForConnection(this.sourceConnectionId);
+		const connectionString = await azdata.connection.getConnectionString(this.sourceConnectionId, true);
 		try {
 			const xEventsFilesFolderPath = '';		// to-do: collect by prompting the user in the UI - for now, blank = disabled
-			const response = (await this.migrationService.getAssessments(ownerUri, this._databasesForAssessment, xEventsFilesFolderPath))!;
+			const response = (await this.migrationService.getAssessments(connectionString, this._databasesForAssessment, xEventsFilesFolderPath))!;
```

Comment:
@cheenamalhotra Have we thought about how extensions are going to handle AAD auth with token refreshes and whatnot when their code lives in a separate service? Seems like something we may want to bundle into this extension host library too...

Reply:
It really depends on how this extension communicates with SqlClient. The host library does seem like a good idea, where we can keep all future MSAL.NET stuff. 

Although as of now, I don't think this extension supports Azure MFA, as far as the code goes, it doesn't seem to pass access token for SQL resource to SqlClient: https://github.com/microsoft/sqltoolsservice/blob/ab5a1e6c859445a050810bd194a1420ac4d03321/src/Microsoft.SqlTools.Migration/MigrationService.cs#L127-L132 

@aasimkhan30 do you plan to use Azure MFA auth with this extension?

As of now, access token seems to be only used for ARM client access (StaticTokenCredential).

Reply:
Are you talking about something like this? 
https://github.com/microsoft/azuredatastudio/blob/a38e8a74c399cbfbaa9b66eed029cdaa920f53b4/extensions/import/src/wizard/pages/summaryPage.ts#L142

Reply:
The backend service is like this https://mssqltools.visualstudio.com/_git/SqlOpsStudioExtensions?path=/services/src/Microsoft.SqlTools.FlatFileImport.Services/FlatFileImportService.cs&version=GBmain&line=150&lineEnd=151&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Is there a better way to implement this? As of now, the extension did not support that anyway. So we are not losing any functionality. I can have a follow up task for this. Thanks

Reply:
Something like that, yeah. But then again this whole extension is meant for migrating TO Azure so this might not even be a scenario that this extension cares about. I was mostly just bringing it up as a heads up for something to think about since plenty of other extensions will want to use MFA connections themselves.

Reply:
As long as you don't need to refresh this access token, it should be fine as of now.
In future, we would be providing APIs in .NET layer from where you can grab access token with account Id or email address + tenant info without the need to use azurecore extension.

Reply:
To chime in here, we do plan on supporting MFA and AAD connections in the future. We have not needed it in the past as our initial 2 migration scenarios were physical migrations (via backup and restore) and so respective RPs directly communicate with each other. 

SQL DB scenarios and login migration scenarios are login migrations, and we need credentials for the target. Currently, we only support SQL creds for the first phase. We have work items to support MFA at the end of Zinc or early Gallium.